tests/unit/materializers/test_numpy_materializer.py

```diff
@@ -0,0 +1,102 @@
+#  Copyright (c) ZenML GmbH 2021. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import os
+from tempfile import TemporaryDirectory
+from typing import Dict
+
+import numpy as np
+
+from tests.unit.test_general import _test_materializer
+from zenml.materializers.numpy_materializer import NumpyMaterializer
+from zenml.metadata.metadata_types import (
+    DType,
+    MetadataType,
+    MetadataTypeTuple,
+)
+
+
+def test_numpy_materializer():
+    """Test the numpy materializer."""
+    arr = np.array([0, 1, 2, "hello", "world", "hello"])
+
+    result = _test_materializer(
+        step_output_type=np.ndarray,
+        materializer_class=NumpyMaterializer,
+        step_output=arr,
+    )
+    assert np.array_equal(arr, result)
+
+
+def test_extract_metadata():
+    """Test the extract_metadata method of the NumpyMaterializer."""
+
+    def _test_extract_metadata(
+        step_output: np.ndarray,
+    ) -> Dict[str, MetadataType]:
+        with TemporaryDirectory() as artifact_uri:
+            materializer = NumpyMaterializer(uri=artifact_uri)
+            existing_files = os.listdir(artifact_uri)
+
+            # Assert that materializer saves something to disk
+            materializer.save(step_output)
+            new_files = os.listdir(artifact_uri)
+            assert len(new_files) > len(
+                existing_files
+            )  # something was written
+
+            # Assert that metadata extraction returns a dict
+            metadata = materializer.extract_metadata(step_output)
+            assert isinstance(metadata, dict)
+            for key, value in metadata.items():
+                assert isinstance(key, str)
+                assert isinstance(value, MetadataTypeTuple)
+
+            return metadata
```

Comment:
This seems duplicated with `_test_materializer()` except for the return. I'd suggest to change `_test_materializer()` instead to make it work with your test. The easiest solutions that come to my mind:
- `return loaded_data, metadata`
- or, take an optional arg `metadata_validation_function` that is then called on the metadata, similar to what the `validation_function` arg does for the loaded data

Reply:
Good idea, i have gone with ading a condition to the `_test_materializer()` to return or not return metadata with loaded data